UPGRADE.md

```diff
@@ -1,10 +1,41 @@
 # Upgrade Guide
 
-_After `7.4.0` all `MINOR` and `MAJOR` version bumps will have upgrade notes 
-posted here._
+_`MAJOR` version bumps will have upgrade notes posted here._
+
+[2020-09-28] 7.x.x to 8.x.x
+-----------------------------
+
+### Overview
+
+Version `8.x.x` is the first version that officially drops support for Java 7. 
+
+- Removal of dependencies offering functionality included in Java 8 and beyond:
+    - [Guava functional interfaces](https://github.com/twilio/twilio-java/pull/574), for example:
+        - `com.google.common.base.Joiner` -> `String.join()`
+        - `com.google.common.collect.Lists` -> `Array.asList()`
+        - `com.google.common.base.Function` -> `java.util.function.Function`
+        - [`com.google.common.base.MoreObjects` -> `lombok.ToString`](https://github.com/twilio/twilio-java/pull/576)
+    - [Guava concurrency, hashing, and charstreams](https://github.com/twilio/twilio-java/pull/575)
+        - `com.google.common.hash.Hash*` -> `org.apache.commons.codec.digest.DigestUtils`
+        - `com.google.common.io.CharStreams` -> `org.apache.commons.io.IOUtils`
+        - `com.google.common.util.concurrent.*` -> `java.util.concurrent`
+    - [Guava `com.google.common.collect.Range` removal]()
```

Comment:
PR link to be added when available

Reply:
https://github.com/twilio/twilio-java/pull/584